### ISSUE TITLE
Create Documentation Website

### ISSUE THREAD

**Issue by krowvin** [USACE-WaterManagement/repgen5#33]
Should we create a documentation website custom or would it be better to implement a github actions process to run pydoc? 

A custom website could allow us to create custom examples with catered documentation. 

Pydoc is more automated and easier to get going to actually document some of the features available now in repgen5 (like the dunder methods for Value)

**Comment by krowvin** [comment]
Got the documentation creating locally and setup the action for it. Action seems to complete but adding the step to write the docs fails. Possibly need this enabled unless someone knows a better way to auto write/generate the docs!

# Settings > Actions > General

<img width="866" alt="image" src="https://github.com/user-attachments/assets/0c70b611-7707-4ab7-8090-18d0baf8ad26">

- [Action Run Error](https://github.com/USACE-WaterManagement/repgen5/actions/runs/10017527171/job/27691823452)

**Comment by MikeNeilson** [comment]
Alternatively, hooking up to readthedocs is a valid option. sphinx was designed to document python so it integrates well with pydoc and has either a familiar format *or* it can be customized.

That's how we're doing the OpenDCS documentation. It's also built as part of the github actions for dev feedback but readthedocs is responsible for the actual rendering and hosting.

**Comment by krowvin** [comment]
Makes sense to me! A little consistency never hurt when you're maintaining multiple documentation sites across enterprises. 

Few questions 

- do we have a business read the docs? 
- how is the login handled for RtD? 
  - Is it just your personal? So that we can link it to our repos docs directory. (Which doesnt exist until we enable that write feature in the screenshot, or we manually re create them each time we make a change...)

**Comment by krowvin** [comment]
Just updating this, we setup a readthedocs site but it needs work

https://repgen.readthedocs.org
